src/core/runtime/dns-nodejs.js

```diff
@@ -3,19 +3,58 @@
 const dns = require('dns')
 
 module.exports = (domain, opts, callback) => {
-  dns.resolveTxt(domain, (err, records) => {
-    if (err) {
-      return callback(err, null)
-    }
+  resolveDnslink(domain)
+    .catch(err => {
+      // If the code is not ENOTFOUND then throw the error
+      if (err.code !== 'ENOTFOUND') throw err
 
-    // TODO: implement recursive option
-
-    for (const record of records) {
-      if (record[0].startsWith('dnslink=')) {
-        return callback(null, record[0].substr(8, record[0].length - 1))
+      if (domain.indexOf('_dnslink') === -1) {
+        // Check the _dnslink subdomain
+        const _dnslinkDomain = ['_dnslink', ...domain.split('.')].join('.')
+        // If this throws then we propagate the error
+        return resolveDnslink(_dnslinkDomain)
+      } else if (domain.split('.').indexOf('_dnslink') === 0) {
+        // The supplied domain contains a _dnslink component
+        // Check the non-_dnslink domain
+        const rootDomain = domain.split('.').slice(1).join('.')
+        return resolveDnslink(rootDomain)
       }
-    }
+      throw err
+    })
+    .then(dnslinkRecord => {
+      callback(null, dnslinkRecord.substr(8, dnslinkRecord.length - 1))
+    })
+    .catch(callback)
+}
 
-    callback(new Error('domain does not have a txt dnslink entry'))
+function resolveDnslink(domain) {
+  const DNSLINK_REGEX = /^dnslink=.+$/
+  return new Promise((resolve, reject) => {
+    dns.resolveTxt(domain, (err, records) => {
+      if (err) return reject(err)
+      resolve(records)
+    })
   })
+    .then(records => {
+      // records is an array of arrays of strings
+      // the below expression flattens it into an array of strings
+      const flatRecords = [].concat(...records)
```

Comment:
We have `lodash` available, shall we use `lodash.flatten` so this isn't quite so cryptic?